test/plots/rectanglePlotTests.ts

```diff
@@ -441,6 +441,80 @@ describe("Plots", () => {
 
         svg.remove();
       });
+
+    });
+
+    describe("Rectangle Plot With Labels", () => {
+      let svg: d3.Selection<void>;
+      let rectanglePlot: Plottable.Plots.Rectangle<number, number>;
+      let DATA: [any];
+      let dataset: Plottable.Dataset;
+      beforeEach(() => {
+        let xScale = new Plottable.Scales.Linear();
+        let yScale = new Plottable.Scales.Linear();
+        svg = TestMethods.generateSVG(SVG_WIDTH / 2, SVG_HEIGHT);
+        rectanglePlot = new Plottable.Plots.Rectangle<number, number>();
+        DATA = [
+          { x: 0, y: 0, x2: 1, y2: 1, val: 1 },
+          { x: 0, y: 1, x2: 1, y2: 2, val: 2 }
+        ];
+        dataset = new Plottable.Dataset(DATA);
+        rectanglePlot.addDataset(dataset);
+        rectanglePlot.x((d: any) => d.x, xScale)
+                     .y((d: any) => d.y, yScale)
+                     .x2((d: any) => d.x2)
+                     .y2((d: any) => d.y2)
+                     .label((d: any) => d.val)
```

Comment:
Theoretically, this shouldn't compile, because `val` is a `number`, not a `string`. However, Typescript doesn't flag this because of the way <any>s work:
```Typescript
(d: any) => d.val; // Identifies as (d: any) => any
```

Not sure what to do about this.